src/ax/saga/mapping/direct/byfield.go

```diff
@@ -42,3 +42,29 @@ func (r byFieldResolver) InstanceIDForMessage(env ax.Envelope) (saga.InstanceID,
 
 	return id, true
 }
+
+// WithPrefix returns a mapper that wraps around m and adds a prefix string to the
+// instance ID returned by m's Resolver.
+//
+// If m's Resolver does not return a valid instance ID, no prefix is added.
+func WithPrefix(prefix string, m *Mapper) *Mapper {
+	return &Mapper{
+		withPrefixResolver{
+			pre: prefix,
+			m:   m,
```

Comment:
Any reason to retain the `Mapper` here, rather than using `m.Resolver` ? I think it might be better to make `withPrefixResolver` wrap another resolver instead of a full mapper. I like the chaining of resolvers though.